### DISCUSSION TITLE
How to list pdf files present in a device?

### DISCUSSION THREAD

**Discussion by GAGANsinghmsitece** [ionic-team/capacitor#4112] [Q&A]
I'm creating a PDF reader using capacitor. When the app starts, I want to show a list of pdf which are currently present on the device of user(whether it's internal or external storage) so that user can select which pdf to open by clicking on it, but I'm unable to retrieve such list. How can I achieve that?

**Comment by jcesarmobile** [comment]
Using Filesystem plugin you can read a folder content, but if you want to scan the whole device for pdf files you'll have to do it recursively for every FilesystemDirectory type.

**Reply by GAGANsinghmsitece** [reply]
I tried following code:-

    async function readdir() {
      try {
        let ret = await Filesystem.readdir({
        path: '/',
        directory: FilesystemDirectory.External
        });
        console.log(ret);
      } catch(e) {
         console.error('Unable to read dir', e);
      } 
    }
    readdir();
but it returns an empty array.
